src/components/missions/MissionsLogic.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import MissionsTable from './MissionsTable';
+import { missionsData } from '../../redux/missions/MissionsAPI';
+
+const MissionsLogic = () => {
+  const { loading, missions, error } = useSelector((state) => state.missions);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(missionsData());
+  }, [dispatch]);
```

Comment:
- [ ] Kindly make sure the fetched data are not being added to the store whenever a user changes view / uses navigations.

    - ### APP BEHAVIOUR
    
    ![chrome-capture-2023-4-18 (3)](https://github.com/AbbasSarwar/SpaceTravelersHub/assets/87197412/af421bfb-bce0-41dd-9083-6129ce3aa239)

    _**Hint: Make sure to dispatch the `missionsData()` action only when the `missions`  array in the redux store is empty**_